Assets/HoloToolkit/Input/Tests/Scripts/ChangeColorOnFocus.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using HoloToolkit.Unity;
+using UnityEngine;
+
+/// <summary>
+/// Controls object appearance by changing its color when focused.
+/// </summary>
+public class ChangeColorOnFocus : Interactable
+{
+    [Tooltip("Object color changes to this when focused.")]
+    [SerializeField]
+    private Color focusedColor = Color.red;
+
+    private Material materialInstance;
+
+    private Color originalColor;
+
+    private void Start()
+    {
+        materialInstance = GetComponent<Renderer>().material;
```

Comment:
I encourage not using this pattern.
1. renderer.material.color will make a copy of the material in memory increasing memory and draw calls
2. renderer.sharedmaterial.color is better than 1 but will change all objects that use that material
3. Best is you use a shader on the material which exposes a highlight color and use this script to change that color. To reference this code, please look at the Holograms 210 Gaze academy code for highlighting objects. It uses a custom shader for this which is more performant than 1 and 2.